sale_order_product_recommendation/static/src/js/units_included_widget.js

```diff
@@ -0,0 +1,56 @@
+odoo.define(
+    "sale_order_product_recommendation.units_included_widget",
+    function (require) {
+        "use strict";
+
+        const AbstractField = require("web.AbstractField");
+        const fieldRegistry = require("web.field_registry");
+
+        const UnitsIncludedWidget = AbstractField.extend({
+            template: "sale_order_product_recommendation.units_included_widget",
+            supportedFieldTypes: ["float"],
+
+            events: {
+                "click .o_units_included_widget_minus": "_onClickMinus",
+                "click .o_units_included_widget_plus": "_onClickPlus",
+                "change .o_product_widget_input": "_onChangeInput",
+                "focus .o_product_widget_input": "_onInputFocus",
+            },
+
+            _updateValue: function (value) {
+                const updatedValue = parseFloat(value) >= 0 ? parseFloat(value) : 0;
+                this._setUpdatedValue(updatedValue);
+            },
+
+            _onClickMinus: function () {
+                this._updateValue(this.value - 1);
+            },
+
+            _onClickPlus: function () {
+                this._updateValue(this.value + 1);
+            },
+
+            _onChangeInput: function (ev) {
+                this._updateValue(ev.target.value);
+            },
+
+            _onInputFocus: function (ev) {
+                ev.target.select();
+            },
+
+            _setUpdatedValue: function (value) {
```

Comment:
Please!! use the method _setValue instead

Reply:
done! i was studying the _setValue function to use it :) thanks!